src/background/Wallet/Wallet.ts

```diff
@@ -1728,30 +1842,119 @@ class PublicController {
+    id,
+    params: { txBase64, clientScope },
     context,
+  }: PublicMethodParams<{
+    txBase64: string;
+    clientScope?: string;
+  }>) {
+    return this.sol_signAndSendTransaction({
+      id,
+      params: { txBase64, clientScope, method: 'signTransaction' },
+      context,
+    });
+  }
+
+  async sol_signAndSendTransaction({
     id,
-    params,
-  }: PublicMethodParams<[] | [{ nonEip6963Request?: boolean }] | null>) {
+    params: { txBase64, clientScope, method = 'signAndSendTransaction' },
+    context,
+  }: PublicMethodParams<{
+    txBase64: string;
+    clientScope?: string;
+    method?: 'signAndSendTransaction' | 'signTransaction';
+  }>) {
+    const currentAddress = this.wallet.ensureCurrentAddress();
+    if (!this.wallet.allowedOrigin(context, currentAddress)) {
+      throw new OriginNotAllowed();
+    }
+    const searchParams = new URLSearchParams({
+      origin: context.origin,
+      transaction: txBase64, // JSON.stringify(transaction),
+      ecosystem: 'solana',
+      method,
+    });
+    if (clientScope) {
+      searchParams.append('clientScope', clientScope);
+    }
+
+    return new Promise((resolve, reject) => {
+      this.safeOpenDialogWindow(context.origin, {
+        requestId: `${context.origin}:${id}`,
+        route: '/sendTransaction',
+        // height: isDeviceWallet ? 800 : undefined,
+        search: `?${searchParams}`,
+        tabId: context.tabId || null,
+        onResolve: (result: SolTransactionResponse) => {
+          console.log({ result });
+          resolve(result);
+        },
+        onDismiss: () => {
+          reject(new UserRejectedTxSignature());
+        },
+      });
+    });
+  }
+
+  async eth_requestAccounts(
+    {
+      context,
+      id,
+      params,
+    }: PublicMethodParams<[] | [{ nonEip6963Request?: boolean }] | null>,
+    opts: { ecosystem: 'ethereum' | 'solana' } = { ecosystem: 'ethereum' }
+  ) {
     if (debugValue && process.env.NODE_ENV === 'development') {
       // eslint-disable-next-line no-console
       console.log('PublicController: eth_requestAccounts', debugValue);
     }
     const currentAddress = this.wallet.readCurrentAddress();
-    if (currentAddress && this.wallet.allowedOrigin(context, currentAddress)) {
-      const { origin } = context;
-      emitter.emit('requestAccountsResolved', {
-        origin,
-        address: currentAddress,
-        explicitly: false,
-      });
-      // Some dapps expect lowercase to be returned, otherwise they crash the moment after connection
-      const result = [currentAddress.toLowerCase()];
-      if (debugValue && process.env.NODE_ENV === 'development') {
-        result.push(String(debugValue));
+    const addrIsEthereum = currentAddress && isEthereumAddress(currentAddress);
+    const addrIsSolana = currentAddress && isSolanaAddress(currentAddress);
+    if (opts.ecosystem === 'ethereum') {
+      invariant(!currentAddress || addrIsEthereum, 'addresss is not ethereum');
```

Comment:
Is there a plan to update this?
And show connect wallet screen in case ecosystem doesn't match selected wallet?